specifyweb/frontend/js_src/lib/components/ExportFeed/Editor.tsx

```diff
@@ -360,14 +362,31 @@ function UserPicker({
   readonly isRequired: boolean;
 }): JSX.Element {
   const resource = React.useMemo(() => new tables.Agent.Resource(), []);
-  React.useEffect(
-    () =>
-      void resource.set(
-        'specifyUser',
-        id === undefined ? null : getResourceApiUrl('SpecifyUser', id)
-      ),
-    [resource, id]
-  );
+  const field = getField(tables.Agent, 'specifyUser');
+
+  const [_, setBlockers] = useSaveBlockers(resource, field);
+
+  React.useEffect(() => {
+    void resource.set(
+      'specifyUser',
+      id === undefined ? null : getResourceApiUrl('SpecifyUser', id)
+    );
+    if (id === undefined) return;
+    // Fetch resource to check for invalid SpecifyUser id entered through XML editor
+    fetchResource('SpecifyUser', id, false).then((res) => {
+      if (res?.id !== id)
+        void resource.set(
+          'specifyUser',
+          res === undefined ? null : getResourceApiUrl('SpecifyUser', id)
+        );
+      if (res === undefined)
+        setBlockers(
+          [resourcesText.invalidSpecifyUser()],
+          getFieldBlockerKey(field, 'invalidSpecifyUser')
+        );
+      else setBlockers([], getFieldBlockerKey(field, 'invalidSpecifyUser'));
+    });
+  }, [id]);
```

Comment:
to be safe, if user id changed, you should disregard the previous request, rather than set a blocker based on a stale value

```suggestion
    fetchResource('SpecifyUser', id, false).then((res) => {
      if(destructorCalled) return;
      if (res?.id !== id)
        void resource.set(
          'specifyUser',
          res === undefined ? null : getResourceApiUrl('SpecifyUser', id)
        );
      if (res === undefined)
        setBlockers(
          [resourcesText.invalidSpecifyUser()],
          getFieldBlockerKey(field, 'invalidSpecifyUser')
        );
      else setBlockers([], getFieldBlockerKey(field, 'invalidSpecifyUser'));
    });
    let destructorCalled = false;
    return ():void => {
      destructorCalled = true
    };
  }, [id]);
```

Reply:
Now that I think of it, instead of writing this code here to handle this one case, I think we should make QueryComboBox itself add the save blocker if resource is not found - that way, this issue is fixed not just for this one place, but for every place where QueryComboBox is used

Reply:
That would be ideal. It was my initial thought as well but it's hard to avoid the 404 page due to the way our ajax fetch is written. Using `fetchResource` with `strict = false` lets us get away with the 404 but `QueryComboBox` does not use it. I think it uses `rgetPromise` instead? I briefly looked into refactoring it to use a similar `strict` param but couldn't figure it out.

I think the current approach is fine as I don't think there are any other places where QCBX can have an invalid resource. If we run into more cases like these we can look into modifying QCBX to handle it.

Reply:
look for usages of hijackBackboneAjax in the code to see how you can make this work with rgetPromise

Reply:
> any other places where QCBX can have an invalid resource

that could be any place if the database was edited by hand, or migration didn't complete correctly

Reply:
I tried it with `hijackBackbone`. I'm able to avoid the 404 and clear the QCBX but it does not trigger an unload protect and the save blocker doesn't show up until you click on the field. I'm gonna open a different PR for that approach so that we at least have this working solution as a hotfix if needed